src/Runtime/Factors/StringFormatOp.cs

```diff
@@ -681,7 +683,7 @@ public class StringFormatOpBase_NoArgumentNames<TStringFormatOp>
         /// <summary>
         /// Maps the number of argument to an array of default argument names ("0", "1" and so on).
         /// </summary>
-        private static readonly Dictionary<int, string[]> ArgumentCountToNames = new Dictionary<int, string[]>();
+        private static readonly ConcurrentDictionary<int, string[]> ArgumentCountToNames = new ConcurrentDictionary<int, string[]>();
```

Comment:
Switch to ReadOnlyArray<>, please. Now that we have this zero-cost abstraction, it's better to use it